File: src/main/java/sample/PaymentRepository.java
package sample;

import org.springframework.r2dbc.core.DatabaseClient;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

@Component
class PaymentRepository {

    final private DatabaseClient databaseClient;

    public PaymentRepository(DatabaseClient databaseClient) {
        this.databaseClient = databaseClient;
    }

    Mono<Integer> save(Payment payment) {
        String q = "INSERT INTO payments (id, account_id, amount) VALUES ($1, $2, $3)";
        return databaseClient.sql(q)
                .bind(0, payment.id)
                .bind(1, payment.accountId)
                .bind(2, payment.amount)
                .fetch()
                .rowsUpdated();
    }
}

File: src/main/kotlin/sample/PaymentRepository.kt
package sample

import org.springframework.data.r2dbc.core.DatabaseClient
import org.springframework.stereotype.Component
import reactor.core.publisher.Mono
import java.sql.Timestamp

@Component
class PaymentRepository(
  private val databaseClient: DatabaseClient
) {

  fun save(payment: Payment): Mono<Int> {
    val q = "INSERT INTO payments (id, account_id, amount) VALUES ($1, $2, $3)"
    return databaseClient.execute(q)
      .bind(0, payment.id)
      .bind(1, payment.accountId)
      .bind(2, payment.amount)
      .fetch()
      .rowsUpdated()
  }
}